### PR TITLE
🎨 set not found PROCESS outputs to NaN instead of 0

### LINKED ISSUE #562: Create plasmod mock output data.
## Description of issue / requirement to address

At present, it is not possible to run plasmod in mock runmode. That would be useful when running tests or developing features for which "example" plasmod output files are necessary.

## Proposed solution

Create at set of plasma output data to be added to the develop branch

### LINKED ISSUE #1039: PROCESS wrapper sets value to 0 if not found
## Describe the bug

We shouldnt modify the output of PROCESS if we can't find a value in the output, we should just set it to `None`

## Steps to reproduce

See here: https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/782d022131095e32d5846faaf027f85c709ff12c/bluemira/codes/process/teardown.py#L205

## Screenshots / Tracebacks

If applicable, add screenshots / tracebacks to help explain your problem.

## Environment (please complete the following information)

- OS: [e.g. Ubuntu 18.04]
- Version [e.g. 0.0.x]
- Branch (if developing) [e.g. develop]

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Closes #1039
Closes #562 

## Description

Instead of setting not found values to 0 (a valid output) set not found values to NaN

## Interface Changes

If a value isnt found in a PROCESS output it will now be set to `np.nan` instead of 0

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
Yeah this is probably a good idea, just need to make sure that it isn't happening for anything important! There are a few of those warnings that come up regularly, but as AFAIK they are for variables we don't use on the EU-DEMO side at least. 

In any case, a NaN is better than a wrong assumption, even if it does cause us the occasional headache.

**Comment by je-cook** [comment]
@CoronelBuendia do we have any specific defaults for PLASMOD we want set? It seems this PR leans into that if we do quite easily

File: bluemira/codes/process/_teardown.py [hunk 1]
```diff
@@ -284,7 +276,7 @@ def _raise_on_infeasible_solution(m_file) -> None:
     error_code = int(m_file.data["ifail"]["scan01"])
     if error_code != 1:
         message = (
-            f"PROCESS did not find a feasible solution. ifail = {error_code}."
+            f"{m_file._name} did not find a feasible solution. ifail = {error_code}."
```
[the remark this answers is not in the document]
Comment (hsaunders1904) [reply bluemira/codes/process/_teardown.py:279]: Also, should we be using a private PROCESS API?

**Review by hsaunders1904 — COMMENTED** [review]
All looks OK to me, just a couple of questions

File: bluemira/codes/process/_teardown.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/codes/process/_teardown.py:279]: its equivalent and the name is translated through the solver. I can make it public if you want (on the mfile)

[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/codes/process/_teardown.py:279]: `_MFileWrapper` that is

[the remark this answers is not in the document]
Comment (hsaunders1904) [reply bluemira/codes/process/_teardown.py:279]: Ah, I see now. Fine with me as-is

**Review by hsaunders1904 — APPROVED** [review]

**Pull request merged**